### PR TITLE
[zh] Resync kubelet configuration

### PR DESCRIPTION
**Pull request by tengqm**
The kubelet reference was out-of-sync, and there have been some new changes in 1.20. This PR does a fresh resync.

### CODE REVIEW

**Comment by howieyuen** [comment]
Every time I see your PR, I have kind of foreboding... cause it is a huge one 😨

**Comment by tengqm** [comment]
@howieyuen It was huge. It took me 4 hours to get it done. I had to run `kubelet --help` on 1.20, cross verify the English version (which is also outdated) and the Chinese version (much older).

**Comment by tengqm** [comment]
preview: https://deploy-preview-25812--kubernetes-io-master-staging.netlify.app/zh/docs/reference/command-line-tools-reference/kubelet/

File: content/zh/docs/reference/command-line-tools-reference/kubelet.md [hunk 1]
```diff
@@ -649,552 +569,831 @@ path to private key (default "key.pem") (DEPRECATED: This is a cadvisor flag tha
...
 <td></td><td style="line-height: 130%; word-wrap: break-word;">
 <!--
-Absolute name of the top level cgroup that is used to manage kubernetes components for which compute resources were reserved via '--kube-reserved' flag. Ex. '/kube-reserved'. [default=''] (DEPRECATED: This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.)
+Absolute name of the top level cgroup that is used to manage kubernetes components for which compute resources were reserved via `--kube-reserved` flag. Ex. `/kube-reserved`. (DEPRECATED: This parameter should be set via the config file specified by the Kubelet's `--config` flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.)
 -->
-给出某个顶层 cgroup 绝对名称，该 cgroup 用于管理带 ‘--kube-reserved’ 标签的 kubernetes 组件的计算资源。例如：‘/kube-reserved’。（默认值为 ‘’）（已弃用：在 --config 指定的配置文件中进行设置。有关更多信息，请参阅 https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/。）
+给出某个顶层 cgroup 绝对名称，该 cgroup 用于管理通过标志 <code>--kube-reserved</code>
+为 kubernetes 组件所预留的计算资源。例如：<code>"/kube-reserved</code>。
```
Comment (howieyuen) [inline content/zh/docs/reference/command-line-tools-reference/kubelet.md:1394]: "/kube-reserved" or /kube-reserved ?

File: content/zh/docs/reference/command-line-tools-reference/kubelet.md [hunk 2]
```diff
@@ -1278,67 +1469,76 @@ If non-empty, write log files in this directory
...
 <!--
-log to standard error instead of files (default true)
+Sets the log format. Permitted formats: `text`, `json`.\nNon-default formats don't honor these flags: `--add-dir-header`, `--alsologtostderr`, `--log-backtrace-at`, `--log_dir`, `--log-file`, `--log-file-max-size`, `--logtostderr`, `--skip_headers`, `--skip_log_headers`, `--stderrthreshold`, `--log-flush-frequency`.\nNon-default choices are currently alpha and subject to change without warning. (DEPRECATED: This parameter should be set via the config file specified by the Kubelet's `--config` flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.)
 -->
-日志输出到 stderr 而不是文件（默认值为 true）
+设置日志文件格式。可以设置的格式有：<code>"text"</code>、<code>"json"</code>。
+非默认的格式不会使用以下标志的配置：<code>--add-dir-header</code>, <code>--alsologtostderr</code>,
+<code>--log-backtrace-at</code>, <code>--log-dir</code>, <code>--log-file</code>,
+<code>--log-file-max-size</code>, <code>--logtostderr</code>, <code>--skip-headers</code>,
+<code>--skip-log-headers</code>, <code>--stderrthreshold</code>, <code>--log-flush-frequency</code>.。
```
Comment (howieyuen) [inline content/zh/docs/reference/command-line-tools-reference/kubelet.md:1512]: extra English dot ('.')

File: content/zh/docs/reference/command-line-tools-reference/kubelet.md [hunk 3]
```diff
@@ -1683,93 +1967,126 @@ If true, when panics occur crash. Intended for testing.
...
+<td colspan="2">--root-dir string&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<!--Default: `/var/lib/kubelet`-->默认值：<code>/var/lib/kubelet</code></td>
 </tr>
 <tr>
 <td></td><td style="line-height: 130%; word-wrap: break-word;">
 <!--
-Directory path for managing kubelet files (volume mounts,etc). (default "/var/lib/kubelet")
+Directory path for managing kubelet files (volume mounts, etc).
 -->
-设置用于管理 kubelet 文件的根目录（例如挂载卷的相关文件）（默认值为 “/var/lib/kubelet”）
+设置用于管理 kubelet 文件的根目录（例如挂载卷的相关文件）。
```
Comment (howieyuen) [inline content/zh/docs/reference/command-line-tools-reference/kubelet.md:2089]: （例如挂载卷的相关文件等）

File: content/zh/docs/reference/command-line-tools-reference/kubelet.md [hunk 4]
```diff
@@ -1822,38 +2149,46 @@ Optional absolute name of cgroups to create and run the runtime in.
...
 </tr>
 <tr>
 <td></td><td style="line-height: 130%; word-wrap: break-word;">
 <!--
-Timeout of all runtime requests except long running request - pull, logs, exec and attach. When timeout exceeded, kubelet will cancel the request, throw out an error and retry later. (default 2m0s) (DEPRECATED: This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.)
+Timeout of all runtime requests except long running request - `pull`, `logs`, `exec` and `attach`. When timeout exceeded, kubelet will cancel the request, throw out an error and retry later. (DEPRECATED: This parameter should be set via the config file specified by the Kubelet's `--config` flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.)
 -->
-除了长时间运行的请求（包括 pull、logs、exec 和 attach 等操作），设置其他请求的超时时间。到达超时时间时，请求会被取消，抛出一个错误并会等待重试。（默认值为 2m0s）（已弃用：在 --config 指定的配置文件中进行设置。有关更多信息，请参阅 https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/。）
+设置长时间运行的请求（包括 <code>pull</code>、<code>logs</code>、<code>exec</code>
```
Comment (howieyuen) [inline content/zh/docs/reference/command-line-tools-reference/kubelet.md:2159]: 设置除了长时间运行的请求

File: content/zh/docs/reference/command-line-tools-reference/kubelet.md [hunk 5]
```diff
@@ -1822,38 +2149,46 @@ Optional absolute name of cgroups to create and run the runtime in.
...
+和 <code>attach</code> 等操作）之外的其他请求的超时时间。
```
Comment (howieyuen) [inline content/zh/docs/reference/command-line-tools-reference/kubelet.md:2160]: 其他运行时请求的超时时间。

**Review by howieyuen — CHANGES_REQUESTED** [review]

**Comment by howieyuen** [comment]
It takes me one hour and a half to finish reviewing~~~hope these are valuable suggestions

File: content/zh/docs/reference/command-line-tools-reference/kubelet.md [hunk 6]
```diff
@@ -313,9 +316,10 @@ If set, any request presenting a client certificate signed by one of the authori
 <tr>
 <td></td><td style="line-height: 130%; word-wrap: break-word;">
 <!--
-The path to the cloud provider configuration file.
+The path to the cloud provider configuration file. Empty string for no configuration file. (DEPRECATED: will be removed in 1.23, in favor of removing cloud providers code from Kubelet.)
 -->
-云服务商的配置文件路径。
+云驱动配置文件的路径。空字符串表示没有配置文件。
```
[the remark this answers is not in the document]
Comment (tengqm) [reply content/zh/docs/reference/command-line-tools-reference/kubelet.md:321]: opt to 云驱动，changed across the whole page.

File: content/zh/docs/reference/command-line-tools-reference/kubelet.md [hunk 7]
```diff
@@ -399,79 +419,63 @@ Domain for this cluster.  If set, kubelet will configure all containers to searc
...
+<td colspan="2">--container-log-max-size string&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<!-- Default: `10Mi`-->默认值：<code>10Mi</code></td>
 </tr>
 <tr>
 <td></td><td style="line-height: 130%; word-wrap: break-word;">
 <!--
-The container runtime to use. Possible values: 'docker', 'remote', 'rkt(deprecated)'. (default "docker")
+&lt;Warning: Beta feature&gt; Set the maximum size (e.g. 10Mi) of container log file before it is rotated. This flag can only be used with `--container-runtime=remote`.  (DEPRECATED: This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.)
 -->
-要使用的容器运行时。目前支持 ‘docker’、‘remote’、‘rkt(已弃用)’。（默认值为 ‘docker’）。
+&lt;警告：beta 特性&gt; 设置容器日志文件在轮换生成新文件时之前的最大值。
```
[the remark this answers is not in the document]
Comment (tengqm) [reply content/zh/docs/reference/command-line-tools-reference/kubelet.md:451]: no. 原文“Set the maximum size (e.g. 10Mi) of container log file before it is rotated.". 这里特意强调 before it is rotated. 日志何时转储（dump 或者 flush）与此无关。可能进程已经在向新的文件写入了，老的文件的数据还有部分位于内存中，尚未落盘。

**Comment by tengqm** [comment]
> It takes me one hour and a half to finish reviewing~~~hope these are valuable suggestions

Thanks for the huge effort. I cannot figure out for how long this page has been outdated. It was a huge effort to sync and to review.

**Pull request merged**